### DISCUSSION TITLE
CI/CD Development Environments on Premises, no Docker

### DISCUSSION THREAD

**Discussion by vytjak** [microsoft/AL-Go#445] [Ideas]
We have a an "in-house" made setup of compilation/deployment scripts to preexisting development servers. We are using full VMs instead of docker images. 
We would like to make use of AL-Go with all the pipelines, but: 
  1) target preexisting dev. servers (NOT Docker images)
  2) alternatively, create/spin-up full VMs for the builds

Wondering how much effort it would be to incorporate that. We are ready to do some PRs.

**Comment by freddydk** [comment]
A lot of optimizations are happening as we speak on the build side.
Currently, the Run-AlPipeline is handling everything in one step.
This will likely be split into 3 steps or maybe even 3 jobs:
- Build, Sign and Test

- Build will be done without the need of docker and performed very fast (once we have Linux support for the compiler, we will support building your apps on linux as well - even faster and lower price).
- Sign needs to be separated out of Run-AlPipeline (I think) due to new requirements from certificate authorities
- Test will need a container, an ACI, an online environment or a VM - the model will probably be semi-pluggable.

At this time, you probably should not spend time on investigating how to change AL-Go to meet your needs.
Once these changes are in - you should investigate whether they make sense for you and you can do stuff then.

**Comment by vytjak** [comment]
Hello Freddy, thank you for your reply. Judging from what's laid out above, it seems we should park the idea of using AL-Go for a while and stick to in-house solution for now. We will mark down AL-Go repo for future reference though.

**Comment by freddydk** [comment]
Yeah - you don't want to be making double-work.
